hack/ci/setup-kind-cluster.sh

```diff
@@ -0,0 +1,232 @@
+
+start_docker_daemon_ci
+
+# Prevent mtu-related timeouts
+echodate "Setting iptables rule to clamp mss to path mtu"
+iptables -t mangle -A POSTROUTING -p tcp --tcp-flags SYN,RST SYN -j TCPMSS --clamp-mss-to-pmtu
+
+# Make debugging a bit better
+echodate "Configuring bash"
+cat << EOF >> ~/.bashrc
+# Gets set to the CI clusters kubeconfig from a preset
+unset KUBECONFIG
+
+cn() {
+  kubectl config set-context --current --namespace=\$1
+}
+
+kubeconfig() {
+  TMP_KUBECONFIG=\$(mktemp);
+  kubectl get secret admin-kubeconfig -o go-template='{{ index .data "kubeconfig" }}' | base64 -d > \$TMP_KUBECONFIG;
+  export KUBECONFIG=\$TMP_KUBECONFIG;
+  cn kube-system
+}
+
+# this alias makes it so that watch can be used with other aliases, like "watch k get pods"
+alias watch='watch '
+alias k=kubectl
+alias ll='ls -lh --file-type --group-directories-first'
+alias lll='ls -lahF --group-directories-first'
+source <(k completion bash )
+source <(k completion bash | sed s/kubectl/k/g)
+EOF
+
+# Find external IP of node where this pod is running
+echodate "Retrieving the external node IP where this pod is scheduled"
+export NODE_NAME=$(kubectl --kubeconfig /etc/kubeconfig/kubeconfig get pods -l prow.k8s.io/id=$PROW_JOB_ID -o jsonpath="{.items..spec.nodeName}")
+export NODE_IP=$(kubectl --kubeconfig /etc/kubeconfig/kubeconfig get node $NODE_NAME -o jsonpath="{.status.addresses[?(@.type=='ExternalIP')].address}")
+
+if [ -z "$NODE_NAME" ] || [ -z "$NODE_IP" ]; then
+  echodate "This script was unable to determine the external IP for kube-apiserver."
+  exit 1
+fi
+
+# Create kind cluster
+TEST_NAME="Create kind cluster"
+
+echodate "Preloading the kindest/node image"
+docker load --input /kindest.tar
+
+echodate "Creating the kind cluster"
+export KUBECONFIG=~/.kube/config
+
+beforeKindCreate=$(nowms)
+
+cat << EOF > kind-config.yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+name: "${KIND_CLUSTER_NAME}"
+networking:
+  apiServerAddress: "0.0.0.0"
+  disableDefaultCNI: true # disable kindnet
+kubeadmConfigPatches:
+- |
+  kind: ClusterConfiguration
+  apiServer:
+    extraArgs:
+      "kubernetes-service-node-port": "31443"
+    certSANs:
+      - localhost
+      - 127.0.0.1
+      - kubernetes
+      - kubernetes.default.svc
+      - kubernetes.default.svc.cluster.local
+      - 0.0.0.0
+      - ${NODE_IP}
+      - ${KIND_CLUSTER_NAME}
+nodes:
+  - role: control-plane
+EOF
+
+if [ -n "${DOCKER_REGISTRY_MIRROR_ADDR:-}" ]; then
+  mirrorHost="$(echo "$DOCKER_REGISTRY_MIRROR_ADDR" | sed 's#http://##' | sed 's#/+$##g')"
+
+  # make the registry mirror available as a socket,
+  # so we can mount it into the kind cluster
+  mkdir -p /mirror
+  socat UNIX-LISTEN:/mirror/mirror.sock,fork,reuseaddr,unlink-early,mode=777 TCP4:$mirrorHost &
+
+  function end_socat_process {
+    echodate "Killing socat docker registry mirror processes..."
+    pkill -e socat
+  }
+  appendTrap end_socat_process EXIT
+
+  cat << EOF | tee -a /etc/environment
```

Comment:
Does this need to use `tee`? A bit above we simply redirect the output, do we need to show this snippet in the buildlog instead?

Reply:
Wrong reference for the file, this will append this in `kind-config.yaml`